### PR TITLE
removed dependency on camel-http4

### PR DESCRIPTION
**Pull request by acoburn**
This addresses https://jira.duraspace.org/browse/FCREPO-1278

removed runtime dependency on camel-http4
add unit tests for new classes
removed code duplication, as per sonar suggestions

### CODE REVIEW

File: src/main/java/org/fcrepo/camel/FcrepoOperationFailedException.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+     * @param url the requested url
+     * @param statusCode the HTTP response code
+     * @param statusText the text corresponding to the status code
+     * @param location a location url
+     * @param responseHeaders a map of the response headers
+     * @param responseBody the response body
+     */
+    public FcrepoOperationFailedException(final URI url, final int statusCode, final String statusText,
+            final URI location, final Map<String, String> responseHeaders, final String responseBody) {
+        super("HTTP operation failed invoking " + url.toString() + " with statusCode: " + statusCode +
```
Comment (awoods) [inline src/main/java/org/fcrepo/camel/FcrepoOperationFailedException.java]: Should we protect ourselves from a NullPointerException on `url.toString()`?

File: src/main/java/org/fcrepo/camel/FcrepoOperationFailedException.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                (location != null ? ", redirectLocation: " + location.toString() : ""));
+        this.url = url;
+        this.statusCode = statusCode;
+        this.statusText = statusText;
```
Comment (awoods) [inline src/main/java/org/fcrepo/camel/FcrepoOperationFailedException.java]: Not used

Comment (acoburn) [inline src/main/java/org/fcrepo/camel/FcrepoOperationFailedException.java]: I kept `statusText` and passed it into `super`. I agree that the others are not necessary.

**Comment by awoods** [comment]
Resolved with: https://github.com/fcrepo4-labs/fcrepo-camel/commit/1d164a1a88e47414d5623f3d8ed491a64d7a01e0

**Pull request closed without merging**